src/parser.ml

```diff
@@ -1,4 +1,6 @@
-let parse buf = Menhir_parser.identifier Lexer.token buf
+let parse buf =
+  try Ok (Menhir_parser.identifier Lexer.token buf) with
+  | Lang.Parser_error s -> Error s
```

Comment:
`Error (Format.sprintf "parser error: %s" s)`

and add an exception `Lang.Lexer_error`